### ISSUE TITLE
Create TypeScript typings

### ISSUE THREAD

**Issue by ProticM** [ProticM/plazar-js#7]

**Comment by ssimontis** [comment]
Do you have someone working on this yet? Saw your listing on dev.to and would be happy to help out with this if no one else is working on it.

**Comment by ProticM** [comment]
@ssimontis No, it's only me :) I don't have anyone working on this. I didn't make any progress because of lack of time. 

Sure, I would use an extra hand here. Did you create any typings before?

**Comment by ssimontis** [comment]
I don't have any major experience creating typings outside of letting the compiler work its magic for me. Because of that, last night I studied up on how ES modules have complicated typings, different approaches to laying out type files based upon solution architecture, and general best practices to make sure I can contribute something productive.

I would consider myself pretty proficient in Typescript, so I am confident I can get it down. I might be able to get a PR in by tomorrow night and you can decide for yourself :)

**Comment by ProticM** [comment]
@ssimontis I was gonna say, sure, go ahead, but you've already forked it :)

You don't have to rush it, create a PR when you're ready, no pressure.

**Comment by ProticM** [comment]
@ssimontis Did you make any progress on this?

**Comment by ssimontis** [comment]
@ProticM 	I have made some progress, but I have had a contract engagement occupying more time than I would like the past week. Tomorrow I think I can submit complete typings. Your message is making me think that none of my commits have been pushed to my forked repo. I'm off to bed in a minute, but first thing tomorrow I will push my progress.

**Comment by ProticM** [comment]
@ssimontis  Yes, I didn't see anything pushed yet.

**Comment by ssimontis** [comment]
I pushed to my forked copy. I am not sure I am doing this right with Lerna...I have never worked with typings in a monorepo structure that was driven by Lerna. I have found two thorough articles that describe how to scaffold the project for Typescript, but they both contradict each other.

https://blog.logrocket.com/setting-up-a-monorepo-with-lerna-for-a-typescript-project-b6a81fe8e4f8/
https://medium.com/@NiGhTTraX/how-to-set-up-a-typescript-monorepo-with-lerna-c6acda7d4559

Unfortunately, I didn't make as much progress as I had hoped today. I have some other obligations today, but I am willing to take another shot this weekend.
